### DISCUSSION TITLE
Hardware

### DISCUSSION THREAD

**Discussion by pwwukpw** [raomin/ESPAltherma#175] [General]
Cannot find M5StickC seems to suggest discontinued?

You could put up an affiliate link or offer a kit ?

I'm not an electronics wizard but keen to get install with my HA. 

Any other hardware  recommended? Interconnect wires to avoid soldering?

Many thanks Peter

**Comment by pwwukpw** [comment]
Also .. can't find 5 pins JST EH 2.5mm connector ! Does that make me stupid ?

**Reply by pwwukpw** [reply]
Think I've found both ???? Better choices in UK or should I order ? 
![Screenshot_20221023-153342_eBay](https://user-images.githubusercontent.com/88422402/197392461-6b4cf77b-ec44-442f-8330-a7c5cb83b92c.jpg)
![Screenshot_20221023-153653_Chrome](https://user-images.githubusercontent.com/88422402/197392540-1a05493c-e188-45a7-88ac-ba9fbbfd9f10.jpg)

**Comment by DieterTHeck** [comment]
Hi,

the M5StickC+ is rather expensive, you don't necessarily have to use this one (it has a display etc...)
I am for instance quite happy with this one... https://shop.m5stack.com/products/m5stamp-c3u-mate-with-pin-headers 
Also lower power consumption, but you need to configure it a bit manually in plattformio.

The Dupont wires will work (but are no JST EH connector). They should not cost much.
I was also not able to get JST EH for reasonable prices here in the EU.
Best plug in Dupont wires into all 5 PINs, then it is more stable in the connector.

**Reply by pwwukpw** [reply]
Dieter .. thank you .. found it locally .. 

I'm a bit worried about the need to manually configure .. as I've no experience in this platform / Linux.. I'd be happy with a higher cost if it increased chance of getting it working ! Many thanks Peter

**Reply by pwwukpw** [reply]
https://shop.m5stack.com/products/m5stamp-c3u-mate-with-pin-headers Also lower power consumption, but you need to configure it a bit manually in plattformio.
> 
Dieter, thank you I'm ordering the mtstamp as all others seemed to be on a long lead time .. would you be able to advise what config changes I'll need to make please .. Ty Peter

**Reply by DieterTHeck** [reply]
> Dieter, thank you I'm ordering the mtstamp as all others seemed to be on a long lead time .. would you be able to advise what config changes I'll need to make please .. 

In plattformio.ini, add the following Environment

```
[env:esp32c3]
platform = 
	https://github.com/platformio/platform-espressif32.git#feature/arduino-idf-master
platform_packages = 
	framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.2
    platformio/tool-esptoolpy @ https://github.com/tasmota/esptool/releases/download/v3.2/esptool-v3.2.zip
framework = arduino
board = esp32dev
board_build.mcu = esp32c3
board_build.partitions = default.csv
board_build.variant = esp32c3
board_build.f_cpu = 160000000L
board_build.f_flash = 80000000L
board_build.flash_mode = dio
board_build.arduino.ldscript = esp32c3_out.ld
build_unflags = 
	-DARDUINO_ESP32_DEV
	-DARDUINO_VARIANT="esp32"
build_flags = 
	-DARDUINO_ESP32C3_DEV
	-DARDUINO_VARIANT="esp32c3"
lib_ldf_mode = deep+
lib_deps = 
	adafruit/Adafruit NeoPixel@^1.10.6
	PubSubClient
	m5stack/M5Unified@^0.0.7
monitor_speed = 115200
monitor_filters = time
upload_resetmethod = nodemcu
upload_speed = 115200
#initial upload
#upload_port = COM9
# OTA upload
#upload_port = 192.168.xxx.yyy
#upload_protocol = espota
#upload_flags =
#   --port=3232
#   --host_port=35709
#   --timeout=30

[env:m5stickc]
[don't insert this, just a reference to existing code]
```

Uncomment the two entries under # OTA upload later, after initial USB flashing for OT update. Insert the IP address of the device.

And select it as ENV.

![grafik](https://user-images.githubusercontent.com/58054042/197475100-b07937a5-9c8f-49cf-961d-81a5b84826b9.png)

**Reply by DieterTHeck** [reply]
All the rest is identical to other devices (you still need to config it, see manual)

**Reply by pwwukpw** [reply]
Dieter .. 

Amazing thank you .. 

Peter

**Comment by DieterTHeck** [comment]
Have a look here what others did, if you like:
https://github.com/raomin/ESPAltherma/issues/17

**Reply by pwwukpw** [reply]
Where can I look. ? 

Ignore that see you kindly popped in a link I'd not seen .

**Comment by pwwukpw** [comment]
@DieterTHeck 
Dieter, 

Code as per above showed up nicely in the enviromemnt bar in PIO 

Re pins ..  on the StampC3 is it 20 & 21 .. i.e. 
`//Default GPIO PINs for Serial2:
#define RX_PIN    20// Pin connected to the TX pin of X10A 
#define TX_PIN    21// Pin connected to the RX pin of X10A
#endif
`
& is the above the correct entry in the setup.h in that case .. 

Thank you so much for any reply :-)
 

?

**Reply by pwwukpw** [reply]
Pleased to say after trial and error managed to get good comms using 

//Default GPIO PINs for Serial2:
#define RX_PIN    19// Pin connected to the TX pin of X10A 
#define TX_PIN    18// Pin connected to the RX pin of X10A 

Really poor documentation on the M5 site .. nothing seems to indicate which GIO ports are linked to Serial 2. 

Thanks to @DieterTHeck for the Environment for the ini .. without that I would never have been able to get this working !

**Comment by DieterTHeck** [comment]
Great that it works. Have you tried OTA already? Regarding documentation - didn't you get this sticker? RX TX is quite well described? 
![M5Stamp-C3U](https://user-images.githubusercontent.com/58054042/198746652-82b5e172-b21a-450b-af99-9c4628834598.jpg) You'd just connect the RX TX from Daikin board there and define 20/21 in the code you quoted. Did you plug the RX TX cables from Daikin board now into 18/19? I think you can use any of them, but he careful when you keep it connected to USB (used for Communication?)
But better not touch a working system 😅

**Reply by DieterTHeck** [reply]
One warning - I had issues after connecting a 3.3v relay to 3.3v out of that device. I think the espaltherma manual suggests 3.3v. I assume the relay killed the voltage converter. I bought a solid state 5v relay connected to 5v out and that works nicely so far.

**Comment by pwwukpw** [comment]
Yes OTA works great !
